File: Leetcode/2109 Adding Spaces to a String.ts
function addSpaces(s: string, spaces: number[]): string {
    const chars = new Array<string>(s.length + spaces.length);
    for (let i = 0, j = 0, k = 0; i < s.length; i++) {
        if (j < spaces.length && i == spaces[j]) {
            j++;
            chars[k++] = " ";
        }
        chars[k++] = s[i];
    }
    return chars.join("");
}

File: Leetcode/2109 Adding Spaces to a String.java
class Solution {
    public String addSpaces(String s, int[] spaces) {
        var chars = new char[s.length() + spaces.length];
        for (int i = 0, j = 0, k = 0; i < s.length(); i++) {
            if (j < spaces.length && i == spaces[j]) {
                j++;
                chars[k++] = ' ';
            }
            chars[k++] = s.charAt(i);
        }
        return new String(chars);
    }
  
    public String addSpaces(String s, int[] spaces) {
        var newString = new StringBuilder(s.length() + spaces.length);
        for (int i = 0, j = 0; i < s.length(); i++) {
            if (j < spaces.length && i == spaces[j]) {
                j++;
                newString.append(' ');
            }
            newString.append(s.charAt(i));
        }
        return newString.toString();
    }
}
